internal/handler/msg_createuser.go

```diff
@@ -29,6 +37,116 @@ func (h *Handler) MsgCreateUser(ctx context.Context, msg *wire.OpMsg) (*wire.OpM
 		return nil, lazyerrors.Error(err)
 	}
 
-	// TODO https://github.com/FerretDB/FerretDB/issues/1491
-	return nil, common.Unimplemented(document, document.Command())
+	dbName, err := common.GetRequiredParam[string](document, "$db")
+	if err != nil {
+		return nil, err
+	}
+
+	if dbName != "$external" && !document.Has("pwd") {
+		return nil, handlererrors.NewCommandErrorMsg(
+			handlererrors.ErrBadValue,
+			"Must provide a 'pwd' field for all user documents, except those with '$external' as the user's source db",
+		)
+	}
+
+	// https://www.mongodb.com/docs/manual/reference/command/createUser/
+
+	var username string
+	username, err = common.GetRequiredParam[string](document, document.Command())
+
+	if err != nil {
+		return nil, err
+	}
+
+	if err = common.UnimplementedNonDefault(document, "customData", func(v any) bool {
+		if v == nil || v == types.Null {
+			return true
+		}
+
+		cd, ok := v.(*types.Document)
+		return ok && cd.Len() == 0
+	}); err != nil {
+		return nil, err
+	}
+
+	if _, err = common.GetRequiredParam[*types.Array](document, "roles"); err != nil {
+		var ce *handlererrors.CommandError
+		if errors.As(err, &ce) && ce.Code() == handlererrors.ErrBadValue {
+			return nil, handlererrors.NewCommandErrorMsg(
+				handlererrors.ErrMissingField,
+				"BSON field 'createUser.roles' is missing but a required field",
+			)
+		}
+
+		return nil, lazyerrors.Error(err)
+	}
+
+	if err = common.UnimplementedNonDefault(document, "roles", func(v any) bool {
+		if v == nil || v == types.Null {
+			return false
+		}
+
+		r, ok := v.(*types.Array)
+		return ok && r.Len() == 0
+	}); err != nil {
+		return nil, err
+	}
+
+	if err = common.UnimplementedNonDefault(document, "digestPassword", func(v any) bool {
+		if v == nil || v == types.Null {
+			return true
+		}
+
+		dp, ok := v.(bool)
+		return ok && dp
+	}); err != nil {
+		return nil, err
+	}
+
+	// NOTE: In MongoDB, the comment field isn't saved in the database, but used for log and profiling.
+	common.Ignored(document, h.L, "pwd", "writeConcern", "authenticationRestrictions", "mechanisms", "comment")
+
+	id := uuid.New()
+	saved := must.NotFail(types.NewDocument(
+		"_id", dbName+"."+username,
+		"credentials", types.MakeDocument(0),
+		"user", username,
+		"db", dbName,
+		"roles", types.MakeArray(0),
+		"userId", types.Binary{Subtype: types.BinaryUUID, B: id[:]},
+	))
+
+	// Users are saved in the "admin" database.
+	adminDB, err := h.b.Database("admin")
+	if err != nil {
+		return nil, lazyerrors.Error(err)
+	}
+
+	users, err := adminDB.Collection("system.users")
```

Comment:
I think `users` can be `nil` (and no error) if `system.users` doesn't exist.

Reply:
I took a look at the code for the Collection method and did a small test calling `.Collection("xuxu")` and it worked fine, rather than returning nil. It looks like it does some magic behind the scenes.

Reply:
ah, you are right! actually, no magic, a particular implementation for SQLite/PG just always returns a collection.

Reply:
> a particular implementation

It is a part of the contract: https://github.com/FerretDB/FerretDB/blob/4de769a5d6fb2f73fa6ab4468c7eda9d0786d619/internal/backends/database.go#L65